src/robot/libdocpkg/specbuilder.py

```diff
@@ -85,9 +85,9 @@ def _create_arguments(self, elem):
             default_elem = arg.find('default')
             if default_elem is not None:
                 spec.defaults[name] = default_elem.text or ''
-            type_elem = arg.find('type')
-            if type_elem is not None:
+            type_elems = arg.findall('type')
+            if type_elems is not None:
```

Comment:
`type_elems` cannot be `None` anymore. It's always a list and may be empty.

The JSON side had exactly the same problem and the fix is the same on both side as well.